numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -654,7 +654,7 @@ npy_divmod@c@(@type@ a, @type@ b, @type@ *modulus)
     }
     else {
         /* if mod is zero ensure correct sign */
-        mod = (b > 0) ? 0.0@c@ : -0.0@c@;
+        mod = npy_copysign@c@(0, b);
```

Comment:
I guess this effectively does not matter (since the result will always be NaN if b is 0). Would be nice to add a test for this path anyway (I assume it is missing right now).
I suppose this might be marginally slower, but I guess it should be very little and it is nicer to read/uniform.

Reply:
Can you elaborate on what path you are referring to? The `b == 0` case is handled at the top of the function.

Reply:
Sorry, was just a silly comment. `b != 0` is guaranteed here I think, so the old comparison was OK.

Reply:
Oh I see what you mean. Yeah, I think you're right. Do you think it's ok to keep this for consistency with the cpython version here https://github.com/python/cpython/blob/6d43f6f081023b680d9db4542d19b9e382149f0a/Objects/floatobject.c#L648 or should I revert it?

Reply:
Yes, I think it is OK. If it was much slower that would be an argument, but I cannot imagine it. I just stumbled on it, because I tried to make a test related to this change.